processor/compile/figure.js

```diff
@@ -1,21 +1,26 @@
-const nodeToString = require('hast-util-to-string');
+const { imgSizeByWidth } = require('../parse/magic-block-parser');
 
 module.exports = function FigureCompiler() {
   const { Compiler } = this;
   const { visitors } = Compiler.prototype;
 
-  visitors.figcaption = function figcaptionCompiler(node) {
-    const caption = nodeToString(node);
-    return `<figcaption>${caption}</figcaption>`;
-  };
-
   visitors.figure = function figureCompiler(node) {
     const [image, caption] = node.children;
-    return `<figure>
 
-${visitors.image.call(this, image)}
+    const img = {
+      image: [image.url, image.title],
+      caption: this.block(caption),
+      sizing: imgSizeByWidth[image.data.hProperties.width],
+    };
+
+    if (image.border) img.border = image.border;
+
+    const block = {
+      images: [img],
+    };
 
-${visitors.figcaption.call(this, caption)}
-</figure>`;
+    return `[block:image]
```

Comment:
Not a huge fan of serializing all images to magic blocks, but I see the necessity. Will this convert markdown syntax images to magic blocks too? And what will happen if/when I straight up write an `<img>` tag?

Reply:
> Not a huge fan of serializing all images to magic blocks, but I see the necessity.

At the very least, we could use the markdown syntax if there is no sizing data or captioning.

> Will this convert markdown syntax images to magic blocks too?

Currently, yes. Since they're represented in `mdast` as the same thing. But, like I said, we could use markdown syntax sometimes?

> And what will happen if/when I straight up write an <img> tag?

That gets parsed into `mdast` as an `html` block, which I think is correct, in the event someone wants to do custom stuff.

Reply:
Ahh, thanks for the details! Do you think forking the image compiler per your suggestion is worth it, or is it just additional complexity? In any case, I'm good merging this as is.

Reply:
I don't mind it too much, since we already support reading both back in.

Of course though, it would be a waste if we come up with a markdown syntax for representing the extra data.